docs/sphinx/user_guide/tutorial/matrix_transpose_local_array.rst

```diff
@@ -202,66 +202,90 @@ Here is the complete version including execution policy and kernel:
+Alternatively, one can allocated dynamic shared memory by specifying the amount in
+the ``RAJA::LaunchParams`` struct. Dynamic shared memory is supported with all
+backends.
+
+As a first example we ilustrate the usage of static shared memory
+and the use of ``RAJA::launch`` tiling methods. RAJA tiling methods
+take a interation space in ``RAJA::tile`` and output tiles which
+the ``RAJA::loop_icount`` method can iterate over and generate
+global and local tile index offsets.  Moreover, lambda expressions for these
+methods will not be invoked for iterations outside the bounds of an iteration
+space when tile dimensions do not divide evenly the size of the iteration space;
+thus, no conditional checks on loop bounds are needed inside inner loops.
+
+A complete RAJA sequential CPU variant with kernel execution policy and
 kernel is:
 
 .. literalinclude:: ../../../../exercises/launch-matrix-transpose-local-array_solution.cpp
    :start-after: // _mattranspose_localarray_raja_start
    :end-before: // _mattranspose_localarray_raja_end
    :language: C++
 
-Here, the ``RAJA::expt::tile`` method is used to create tilings of the outer 
-'row' and 'col' iteration spaces. The ``RAJA::expt::tile`` method
-takes an additional argument specifying the tile size for the corresponding 
-loop. To traverse the tile, we use the ``RAJA::expt::loop_icount`` method, 
-which is similar to the ``RAJA::ForICount`` statement used in a 
-``RAJA::kernel`` execution policy as shown above. A 
-``RAJA::expt::loop_icount`` method call
+Here, the ``RAJA::tile`` method is used to create tilings of the outer
+'row' and 'col' iteration spaces. The ``RAJA::tile`` method
+takes an additional argument specifying the tile size for the corresponding
+loop. To traverse the tile, we use the ``RAJA::loop_icount`` method,
+which is similar to the ``RAJA::ForICount`` statement used in a
+``RAJA::kernel`` execution policy as shown above. A
+``RAJA::loop_icount`` method call
 will generate local tile index associated with the outer global index.
-The local tile index is necessary as we use it to read and write entries 
+The local tile index is necessary as we use it to read and write entries
 from/to global memory to ``RAJA_TEAM_SHARED`` memory array.
 
+As an alternative to static shared memory, the matrix transpose kernel may be
+express using dynamic shared memory. Prior to invoking the amount of shared memory
+must be specified
 
+.. literalinclude:: ../../../../examples/dynamic_mat_transpose.cpp
+   :start-after: // _dyanamic_mattranspose_shared_mem_start
+   :end-before: // _dyanamic_mattranspose_shared_mem_end
+   :language: C++
+
+The amount of shared memory is then specifed in the ``RAJA::LaunchParams`` struct
+and then accessed within the kernel using the LaunchContext ``getSharedMemory`` method.
+The ``getSharedMemory`` method may be invoked multiple times, each time returning an
+offset to the shared memory buffer. Since the launch context uses a bump style allocator
```

Comment:
```suggestion
offset to the shared memory buffer. Since the launch context uses a bump style allocator,
```

Does the term 'bump style' allocator need to be defined? It may be a foreign concept to many.